openlcs/packages/mixins.py

```diff
@@ -210,3 +254,93 @@ def save_scan_result(self, **kwargs):
+            arch=component_data.get("arch"),
+            defaults={
+                "summary_license": component_data.get("license"),
+            },
+        )
+        return component
+
+    def build_release_node(self):
+        """
+        Build release node. For container, if exit release data, will create
+        release product tree node, then create a parent product tree node,
+        then create some child component product tree node. The parent and
+        child components will be component instances.
+        """
+        # Create container parent components
+        container_component = self.create_component(
+            self.components.get('container_component')
+        )
+        # Create release node
+        release_ctype = ContentType.objects.get_for_model(Release)
+        release_node, _ = ProductTreeNode.objects.get_or_create(
+            name=self.release.name,
+            content_type=release_ctype,
+            object_id=self.release.id,
+            parent=None,
+        )
+        # Create container node
+        cnode, _,  = container_component.release_nodes.get_or_create(
+            name=container_component.name,
+            parent=release_node,
+        )
+        # Create container child components
+        for component_data in self.components.get("components"):
+            component = self.create_component(component_data)
+            component.release_nodes.get_or_create(
+                name=component.name,
+                parent=cnode,
+            )
+
+    def build_container_node(self):
+        """
+        Build container node. For container, will create a parent component
+        tree node, then create some child component tree node. The parent and
+        child components will be component instances.
+        """
+        # Create container parent components
+        container_component = self.create_component(
+            self.components.get('container_component')
+        )
+        # Create container node
+        component_ctype = ContentType.objects.get_for_model(Component)
+        cnode, _ = ComponentTreeNode.objects.get_or_create(
+            name=container_component.name,
+            content_type=component_ctype,
+            object_id=container_component.id,
+            parent=None,
+        )
+        # Create container child components
+        for component_data in self.components.get("components"):
+            component = self.create_component(component_data)
+            ComponentTreeNode.objects.get_or_create(
+                name=component.name,
+                parent=cnode,
+                content_type=component_ctype,
+                object_id=component.id,
+            )
+
+    def save_container_components(self, **kwargs):
+        self.components = kwargs.pop('components')
+        product_release = kwargs.pop('product_release')
+        if product_release:
+            self.release = Release.objects.filter(name=product_release)[0]
```

Comment:
this assumes `product_release` filter exists. in case it does not, there'll be unexpected exceptions while building nodes.

Reply:
the product and release created in create_product_release when import